server/src/test/java/org/opensearch/index/engine/EngineConfigTests.java

```diff
@@ -34,31 +33,29 @@ public void setUp() throws Exception {
     }
 
     public void testEngineConfig_DefaultValueForReadOnlyEngine() {
-        EngineConfig config = new EngineConfig(
-            null,
-            null,
-            defaultIndexSettings,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            null,
-            () -> RetentionLeases.EMPTY,
-            null,
-            null
-        );
+        EngineConfig config = new EngineConfig.Builder().setShardId(null)
+            .setThreadPool(null)
+            .setIndexSettings(defaultIndexSettings)
+            .setWarmer(null)
+            .setStore(null)
+            .setMergePolicy(null)
+            .setAnalyzer(null)
+            .setSimilarity(null)
+            .setCodecService(null)
+            .setEventListener(null)
+            .setQueryCache(null)
+            .setQueryCachingPolicy(null)
+            .setTranslogConfig(null)
+            .setFlushMergesAfter(null)
+            .setExternalRefreshListener(null)
+            .setInternalRefreshListener(null)
+            .setIndexSort(null)
+            .setCircuitBreakerService(null)
+            .setGlobalCheckpointSupplier(null)
+            .setRetentionLeasesSupplier(() -> RetentionLeases.EMPTY)
+            .setPrimaryTermSupplier(null)
+            .setTombstoneDocSupplier(null)
```

Comment:
We dont' need to use setXXX for settings `null` values as it is default. This is one of the advantages of using `builder` pattern as it replaces large constructor with default values.

Reply:
Thanks @dreamer-89 . I used IntelliJ's `Replace constructor with builder` and did not pay much attention to these optimizations. Will take care of these in next commit.